### PR TITLE
Added Logging support using fern

### LINKED ISSUE #408: Real logging in xi-core (+ core RPC logging)
Currently xi-core's only logging mechanism is printing to stderr. We will eventually want to move to a proper logging solution; a good example of this is available in the [xi-tui](https://github.com/little-dude/xi-tui/blob/master/src/main.rs) repo.

In addition to this, a small thing we've talked about in the past is adding complete logging of RPC messages, which would be a useful tool for debugging.

### PR DESCRIPTION
**Pull request by betterclever**
Fixes #408

### CODE REVIEW

**Comment by cmyr** [comment]
So it looks like the approach we'll have to take here is to keep `fern` behind a feature flag, and not expose it on fuchsia. For fuchsia, we will declare our own versions of these macros that just print to stderr.

**Comment by betterclever** [comment]
@cmyr You may have a look. I don't know how to handle it for Fuschia yet. You said something about cfg flags.

**Comment by cmyr** [comment]
Awesome, will need to give this a spin but it looks good.

It looks like Fuchsia won't be a problem; because `fern` is only used in the main binary, which fuchsia doesn't use, there's no dependency issue. In the fuchsia binary we can pick use the fuchsia system logger, which will be a nice improvement there as well, so this looks like a clean win.

File: rust/plugin-lib/src/lib.rs [hunk 1]
```diff
@@ -165,7 +168,7 @@ pub enum Error {
 }
 
 /// Run `plugin` until it exits, blocking the current thread.
-pub fn mainloop<P: Plugin>(plugin: &mut P) -> Result<(), ReadError> {
+pub fn mainloop<P: Plugin>(plugin: &mut P) -> Result<(), ReadError> {
```
Comment (cmyr) [inline rust/plugin-lib/src/lib.rs]: whitespace

File: rust/rpc/Cargo.toml [hunk 2]
```diff
@@ -13,3 +13,4 @@ serde_derive = "1.0"
 crossbeam = "0.3.2"
 
 xi-trace = { path = "../trace", version = "0.1.0" }
+log = "0.4.3"
```
Comment (cmyr) [inline rust/rpc/Cargo.toml]: ditto the earlier cargo comment

File: rust/src/main.rs [hunk 3]
```diff
@@ -15,6 +15,11 @@ use std::io;
 
 extern crate xi_core_lib;
 extern crate xi_rpc;
+extern crate fern;
```
Comment (cmyr) [inline rust/src/main.rs]: again, by convention (and also by alphabetical ordering) I'd prefer the non-xi crates above the internal crates

**Comment by betterclever** [comment]
Done.

**Comment by cmyr** [comment]
Thanks @betterclever, this was long overdue. 😄

**Pull request merged**